### PR TITLE
Creation of parser.jl, with the parser function writed

### LINKED ISSUE #2: Write parser.jl
Write a parser that reads a data instance in `data/Instances_A/foldername/`.
Input: string "A", "B" or "X" and string for the data instance name.
Data instance can be found [here](http://www.roadef.org/challenge/2005/fr/sujet.php).

### PR DESCRIPTION
**Pull request by Jonathan-Fontaine44**
Closes #2

### CODE REVIEW

**Review by remi-garcia — CHANGES_REQUESTED** [review]
`Int64`/`Int`
`Array{T, 1}`/`Vector{T}`
`Array{T, 2}`/`Matrix{T}`

I think we should follow the [JuMP style guide](https://www.juliaopt.org/JuMP.jl/stable/style/)

File: src/parser.jl [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+using CSV
+
+# Library importing the DataFrame type that is easier to manipulate.
+using DataFrames
+
+# Constant of the problem
+OPTIMISATION_FILE_NAME = "optimization_objectives.txt"
+PAINT_FILE_NAME = "paint_batch_limit.txt"
+RATIO_FILE_NAME = "ratios.txt"
+VEHICLES_FILE_NAME = "vehicles.txt"
```
Comment (remi-garcia) [inline src/parser.jl]: Constant/Constants
`CONSTANT_NAME`/`const CONSTANT_NAME`

File: src/parser.jl [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+# An Instance structure that is used to format data as we want.
+struct Instances
+    # objectif function
+    HPRC_rank::Int64
+    LPRC_rank::Int64 # Si il n'y a pas de LPRC, LPRC_rank = -1
```
Comment (remi-garcia) [inline src/parser.jl]: Comments are in English for consistency

File: src/parser.jl [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    nb_HPRC::Int64
+    nb_LPRC::Int64
+    # sequence vehicle data
+    HPRC_flag::Array{Bool, 2}
+    LPRC_flag::Array{Bool, 2}
+    color_code::Array{Int64, 1}
+end
+
+# This function is used to read data of an instance from all files, and agregate into an Instance structure.
+function parser(Instance_name::String, instanceType::String, pathFolder::String="../data/Instances_")
```
Comment (remi-garcia) [inline src/parser.jl]: `Instance_name`/`instance_name`
`instanceType`/`instance_type`
`pathFolder`/`path_folder`

File: src/parser.jl [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        LPRC_flag = Array{Bool, 2}()
+    end
+
+    color_code = Array{Int64, 1}(df_vehicles[!, 4])
+
+    return Instances(HPRC_rank, LPRC_rank, PCB_rank,
+            nb_paint_limitation,
+            HPRC, LPRC, nb_high, nb_low,
+            HPRC_flag, LPRC_flag, color_code
+        )
```
Comment (remi-garcia) [inline src/parser.jl]: Should be
```Julia
    return Instances(HPRC_rank, LPRC_rank, PCB_rank,
                     nb_paint_limitation,
                     HPRC, LPRC, nb_high, nb_low,
                     HPRC_flag, LPRC_flag, color_code
        )
```

**Review by remi-garcia — CHANGES_REQUESTED** [review]
Except for the naming style we just discuss after you opened this PR this looks fine for me.

**Comment by k-fretaud** [comment]
> Int64 -> Int 

May be a good choice (must be check) but I'm not for the Vector/Matrix change

**Comment by remi-garcia** [comment]
According to [this thread](https://groups.google.com/forum/#!topic/julia-users/qs3lZj7AaY8) we should use `Int` since `Int64` isn't necessary.
After this change and the constant one I'll approve this.

**Review by remi-garcia — APPROVED** [review]

**Review by BenoitLoger — APPROVED** [review]

**Pull request merged**